File: Maximum no of 1's row - GFG/maximum-no-of-1s-row.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution
{
    public:
        int maxOnes (vector <vector <int>> &Mat, int N, int M)
        {
            int ans = -1;
            int maxC =-1;
            for(int i=0;i<N;i++){
                if(Mat[i][0]==1){
                    return i;
                }
                int currC = count(Mat[i],0,M-1,M); 
                if(currC>maxC){
                    ans = i;
                    maxC=currC;
                }
            }
            return ans;
        }
        int count(vector<int> arr,int l,int r,int M){
            if(l>r){
                return 0;
            }
            int m = l +(r-l)/2;
            if(m>0 && arr[m-1]==0 && arr[m-1]==1){
                return M-m;
            }
            if(m+1<=M-1&&arr[m]==0 &&arr[m+1]==1){
                return M-m-1;
            }
            if(arr[m]==0){
                return count(arr,m+1,r,M);
            }
            return count(arr,l,m-1,M);
        }
};

//{ Driver Code Starts.

int main(){
    int t; cin >> t;
    while (t--){
        int n, m; cin >> n >> m;
        vector <vector <int>> arr (n, vector <int> (m));
        for (int i = 0; i < n; i++)
            for (int j = 0; j < m; j++)
                cin >> arr[i][j];
        Solution ob;        
        cout << ob.maxOnes(arr, n, m) << endl;
    }
}
// } Driver Code Ends

File: Maximum no of 1's row - GFG/maximum-no-of-1s-row.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;
class GfG
{
    public static void main (String[] args)
    {
        
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int m = sc.nextInt();
            int [][]arr = new int[n][m];
            
            for (int i = 0; i < n; ++i)
                for (int j = 0; j < m; ++j)
                    arr[i][j] = sc.nextInt ();

    		System.out.println (new Sol().maxOnes (arr, n, m));
        }
        
    }
}
// } Driver Code Ends


//User function Template for Java

class Sol
{
    public static int maxOnes (int Mat[][], int N, int M)
    {
        int i=0;
        int j=0;
        int ans = -1;
        int maxC =-1;
        while(i<N&&j<M){
            while(j<M&&Mat[i][j]==0){
                j++;
            }
            if(M-j>maxC){
                ans = i;
                maxC = M-j;
            }
            if(j==M){
                j=0;
            }
            i++;
            while(i<N && j>=0&&Mat[i][j]==1){
                j--;
            }
            if(j==-1){
                j=0;
            }
        }
        return ans;
    }
}